omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/visit/ParserEncounterIntoSimpleObjects.java

```diff
@@ -45,6 +49,9 @@
     private LocationService locationService;
 
     private DispositionService dispositionService;
+    
+    @Autowired
+    DiagnosisServiceCompatibility diagnosisServiceCompatibility;
```

Comment:
hi @dkayiwa,as i want to use getPrimaryDiagnoses(encounter), i followed the link you dropped me and accomplished the following steps.
   1) created an interface called DiagnosisServiceCompatibility

  2)I implemented this in a submodule,code can be seen in DiagnosisServiceCompatibility2_2 class

3) I then tried to autowire the interface as see here,but i run into  https://pastebin.com/Faj3eLCL on building locally.

Could i be missing out some thing in my poms?